test/system/520-checkpoint.bats

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bats   -*- bats -*-
+#
+# test podman checkpoint. Similar in many ways to our pause tests.
+#
+
+load helpers
+
+CHECKED_ROOTLESS=
+function setup() {
+    # FIXME: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1857257
+    # TL;DR they keep fixing it then breaking it again. There's a test we
+    # could run to see if it's fixed, but it's way too complicated. Since
+    # integration tests also skip checkpoint tests on Ubuntu, do the same here.
+    if grep -qiw ubuntu /etc/os-release; then
+        skip "FIXME: checkpointing broken in Ubuntu 2004, 2104, 2110, ..."
+    fi
+
+    # None of these tests work rootless....
+    if is_rootless; then
+        # ...however, is that a genuine cast-in-stone limitation, or one
+        # that can some day be fixed? If one day some PR removes that
+        # restriction, fail loudly here, so the developer can enable tests.
+        if [[ -n "$CHECKED_ROOTLESS" ]]; then
+            run_podman '?' container checkpoint -l
+            is "$output" "Error: checkpointing a container requires root" \
+               "Confirming that rootless checkpoint doesn't work. If that changed, please reexamine this test file!"
+            CHECKED_ROOTLESS=y
+        fi
+        skip "checkpoint does not work rootless"
+    fi
+
+    basic_setup
+}
+
+function teardown() {
+    run_podman '?' volume rm myvol
+
+    basic_teardown
+}
+
+@test "podman checkpoint - basic test" {
+    run_podman run -d $IMAGE sh -c 'while :;do cat /proc/uptime; sleep 0.1;done'
+    local cid="$output"
+
+    # Wait for container to start emitting output
+    wait_for_output '[1-9]\+' $cid
+
+    # Checkpoint, and confirm via inspect
+    run_podman container checkpoint $cid
+    is "$output" "$cid" "podman container checkpoint"
+
+    run_podman container inspect \
+               --format '{{.State.Status}}:{{.State.Running}}:{{.State.Paused}}:{{.State.Checkpointed}}' $cid
+    is "$output" "exited:false:false:true" "State. Status:Running:Pause:Checkpointed"
+
+    # Plan A was to do something similar to 080-pause.bats: sleep for long
+    # enough to cause a gap in the timestamps in the log. But checkpoint
+    # doesn't seem to work like that: upon restore, even if we sleep a long
+    # time, the newly-started container seems to pick back up close to
+    # where it left off. (Maybe it's something about /proc/uptime?)
```

Comment:
I'm not sure if this answers your question, but CRIU supports checkpoint/restore of time namespace.
`crun` supports timens (https://github.com/containers/crun/pull/743), but `runc` does not yet (https://github.com/opencontainers/runc/issues/2345).

Reply:
@rst0git it will take me some time to figure out if that explains what I was seeing, but it looks very likely and is a great lead. Thank you!